Update msal-node caching doc for distributed cache (#5559)
AzureAD/microsoft-authentication-library-for-js: lib/msal-node/docs/caching.md

});
```

In production, you would most likely want to serialize and persist the token cache. Depending on the type of application, you can:

* Desktop apps, console apps (public clients):
  * Use [MSAL Node Extensions](../../../extensions/msal-node-extensions/README.md), which provides persistence and encryption at rest solutions on Windows, Linux and Mac OS
* Web apps, web APIs, daemon apps (confidential client):
  * MSAL's in-memory token cache does not scale for production. Use the [distributed token caching](#performance-and-security) pattern to persist the cache in your choice of storage environment (Redis, MongoDB, SQL databases etc. -keep in mind that you can use these in tandem *e.g.* a Redis-like memory cache as a first layer of persistence, and a SQL database as a second, more stable persistence layer)

## Cache in memory

> 

Reviewer: Please add a note that the default in memory cache is not scalable and performance will degrade after holding a few 100s tokens. For web site / web api scenarios this means a few 100 users. For app 2 app scenario, this means a few 100 tenants.

Author: done!

* For multi-tenant daemon apps using client credentials grant: **tenantId** (or rather, `<clientId>.<tenantId>`)
* For web APIs calling other web APIs using OBO: hash of the incoming access token from the user (i.e. the token which will subsequently be exchanged for an OBO token)

MSAL Node provides the [DistributedCachePlugin](https://azuread.github.io/microsoft-authentication-library-for-js/ref/classes/_azure_msal_node.distributedcacheplugin.html) class for confidential clients, which implements the [ICachePlugin](https://azuread.github.io/microsoft-authentication-library-for-js/ref/interfaces/_azure_msal_common.icacheplugin.html). An instance of `DistributedCachePlugin` requires:

> * a client interface, which implements `get` and `set` operations on the persistence server (Redis, MySQL etc.).

Reviewer 2: ```suggestion
* a **client interface**, which implements `get` and `set` operations on the persistence server (Redis, MySQL etc.).
```

> * a partition manager, for reading from and writing to cache with respect to a given **partition key**.

Reviewer 2: ```suggestion
* a **partition manager**, for reading from and writing to cache with respect to a given **partition key**.
```

> Please refer to the [RedisTestApp](../../../samples/msal-node-samples/RedisTestApp/README.md) for an implementation.

Reviewer 2: ```suggestion
Please refer to the [RedisTestApp](../../../samples/msal-node-samples/RedisTestApp/README.md) for a sample implementation.
```
